### PR TITLE
Removed `prefix` from column constraints

### PR DESCRIPTION
**Pull request by trueqbit**
Introduces the checking of elements and constraints that are passed to the table factory functions.

### CODE REVIEW

File: dev/constraints.h [hunk 1]
```diff
@@ -481,7 +481,6 @@ namespace sqlite_orm {
                                                              check_if_is_type<null_t>,
                                                              check_if_is_type<not_null_t>,
                                                              check_if_is_type<unindexed_t>,
-                                                             check_if_is_template<prefix_t>,
```
Comment (fnc12) [inline dev/constraints.h:484]: are you sure about that? the template is named `is_constraint` not `is_column_constraint`. There are two types of constraints: table constraints and column constraints. E.g. FOREIGN KEY is table constraint and it exists in this template. So I doubt we need to remove `prefix_t` from this list

Comment (trueqbit) [reply dev/constraints.h:484]: I had in mind that `is_constraint` is only ever used to check column constraints in `make_column()`.
However, your point is fair enough. See next commit.

File: dev/constraints.h [hunk 2]
```diff
@@ -501,32 +499,35 @@ namespace sqlite_orm {
...
     template<class... Cs>
     internal::foreign_key_intermediate_t<Cs...> foreign_key(Cs... columns) {
-        return {std::make_tuple(std::forward<Cs>(columns)...)};
+        return {{std::forward<Cs>(columns)...}};
     }
 #endif
 
     /**
-     *  UNIQUE constraint builder function.
+     *  UNIQUE table constraint builder function.
```
Comment (fnc12) [inline dev/constraints.h:515]: `UNIQUE` is also can be applied at table level. I'd remove `table` word from here. More info [here](https://www.sqlite.org/lang_createtable.html)

File: dev/schema/table.h [hunk 3]
```diff
@@ -402,11 +412,17 @@ namespace sqlite_orm {
 
     template<class... Cs, class T = typename std::tuple_element_t<0, std::tuple<Cs...>>::object_type>
     internal::using_fts5_t<T, Cs...> using_fts5(Cs... columns) {
+        static_assert(polyfill::conjunction_v<internal::is_table_element_or_constraint<Cs>...>,
```
Comment (fnc12) [inline dev/schema/table.h:415]: these checks are really useful. Thanks

File: dev/constraints.h (hunk 2 above)
Comment (trueqbit) [reply dev/constraints.h:515]: There are two overloads, see below.

Comment (fnc12) [reply dev/constraints.h:515]: oh I see. Pardon

**Review by fnc12 — APPROVED** [review]

**Pull request merged**